models/Task.php

```diff
@@ -1037,8 +1048,43 @@ public function removeUser($userId)
 
     public function getColor()
     {
-        if($this->task_list_id && $this->list) {
+        if ($this->task_list_id && $this->list) {
             return $this->list->getColor();
         }
     }
+
+    // ###########  handle task related account  ###########
+
+    public function manageTaskAccount()
+    {
+        $accountTitle = "Task#$this->id ( $this->title )";
+        $accountUserId = empty($this->responsibleUsers) ? null : User::findOne(['guid' => $this->responsibleUsers[0]])->id;
+
+        if (null === ($taskAccount = TaskAccount::findOne(['task_id' => $this->id]))) {
+
+            $account = new Account([
+                'title' => $accountTitle,
+                'space_id' => $this->content->container->id,
+                'account_type' => Account::TYPE_TASK,
+                'user_id' => $accountUserId
+            ]);
+
+            $account->save();
+
+            $taskAccount = new TaskAccount([
+                'task_id' => $this->id,
+                'account_id' => $account->id,
+                'account_type' => self::ACCOUNT_SPACE,
+            ]);
+
+            return $taskAccount->save();
+        } else {
+            $account = Account::findOne(['id' => $taskAccount->account_id]);
+
+            $account->title = $accountTitle;
+            $account->user_id = $accountUserId;
+
+            return $account->save();
+        }
+    }
```

Comment:
:+1: lgtm

do we need to worry about if something goes wrong while saving the acount (e.g.)?

Reply:
I don't think so for now, otherwise if something happened when creating the account , the task will be created anyway and we can recreate the account by editing the task.

Reply:
ok